src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -91,20 +144,69 @@ public Atlas slice()
         sliceLines();
         slicePoints();
 
-        // Apply changes and rebuild the Atlas with the changes before slicing Relations
-        final RawAtlasChangeSetBuilder changeBuilder = new RawAtlasChangeSetBuilder(this.rawAtlas,
-                this.afterSlicingLinesAndPoints);
-        final Atlas withSlicedWaysAndPoints = changeBuilder.applyChanges();
-        this.rawAtlas = withSlicedWaysAndPoints;
+        // Apply changes and rebuild the atlas with the changes before slicing relations
+        final ChangeSetHandler simpleChangeBuilder = new SimpleChangeSetHandler(this.rawAtlas,
+                this.slicedPointAndLineChanges);
+        final Atlas atlasWithSlicedWaysAndPoints = simpleChangeBuilder.applyChanges();
+        this.rawAtlas = atlasWithSlicedWaysAndPoints;
```

Comment:
is there a reason why rawAtlas is not final (it seems to have lost its value after the first round of slicing)?  A follow up question is, why is rawAtlas kept in an intermediate state of having only slicedWaysAndPoints and not relations?

Reply:
It's not final because we want to rebuild the atlas after applying the changes from the way/point slicing. The reason why it's separated into two is because slicing ways has the potential to create many new ways and has the implications on relations (deleted the original member and added 2 or more sliced lines that used to make up the deleted member). Dealing with this many number of changes without them being in the Atlas was 1. cumbersome 2. not very efficient when we wanted to make use of Atlas geospatial queries for ways that were sliced but not part of the atlas. Currently in the process of splitting this class up some to separate behavior between the point/way and relation slicing. This may address your concern of non-final.